lib/kafka/produce_operation.rb

```diff
@@ -137,6 +137,9 @@ def handle_response(response)
               delay: ack_time - message.create_time,
             })
           end
+        rescue Kafka::MessageSizeTooLarge
+          @logger.error "Message is too large when writing to #{topic}/#{partition}"
+          @buffer.clear_messages(topic: topic, partition: partition)
```

Comment:
This clears _all_ messages, which causes more data loss than is strictly necessary... however, I think it's a good enough solution for now. Can you add a description to the log message indicating that you're clearing _n_ messages?